src/Compilers/CSharp/Test/Semantic/FlowAnalysis/LocalFunctions.cs

```diff
@@ -1147,5 +1150,173 @@ static void Main(object[] args)
                 //                 Foo();
                 Diagnostic(ErrorCode.ERR_UseDefViolation, "Foo()").WithArguments("x").WithLocation(9, 17));
         }
+
+        [Fact]
+        [WorkItem(14097, "https://github.com/dotnet/roslyn/issues/14097")]
+        public void PiecewiseStructAssign()
+        {
+            var comp = CreateCompilationWithMscorlib(@"
+struct S { public int X, Y; }
+
```

Comment:
It may be worth adding tests with tuples. 
They are also structs allowing field-wise assignment, but also via different aliases ( A, Item1, Rest,...)